File: workspace/richturner/WebConsole_2_0_0_template/src/org/openremote/web/console/panel/entity/component/Label.java
/* OpenRemote, the Home of the Digital Home.
* Copyright 2008-2010, OpenRemote Inc.
*
* See the contributors.txt file in the distribution for a
* full listing of individual contributors.
*
* This program is free software: you can redistribute it and/or modify
* it under the terms of the GNU Affero General Public License as
* published by the Free Software Foundation, either version 3 of the
* License, or (at your option) any later version.
*
* This program is distributed in the hope that it will be useful,
* but WITHOUT ANY WARRANTY; without even the implied warranty of
* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
* GNU Affero General Public License for more details.
*
* You should have received a copy of the GNU Affero General Public License
* along with this program. If not, see <http://www.gnu.org/licenses/>.
*/
package org.openremote.web.console.panel.entity.component;

import org.openremote.web.console.panel.entity.Link;

/**
 * The label can set font size and color, change text by polling status.
 */
@SuppressWarnings("serial")
public class Label extends Component {

   private int fontSize;
   private String color;
   private String text;
   private Link link;
   
   public Label() {
   }
   
   public int getFontSize() {
      return fontSize;
   }
   
   public void setLink(int fontSize) {
   	this.fontSize = fontSize;
   }
   
   public String getColor() {
      return color;
   }
   
   public void setColour(String color) {
   	this.color = color;
   }
   
   public String getText() {
      return text;
   }

   public void setText(String text) {
   	this.text = text;
   }
   
   public Link getLink() {
   	return link;
   }
   
   public void setLink(Link link) {
   	this.link = link;
   }
}

File: attic/tags/OpenRemote_Boss_2_0_0_Alpha8/iPhone_Console_2_0_0_Beta1/Classes/Label.m
/* OpenRemote, the Home of the Digital Home.
 * Copyright 2008-2009, OpenRemote Inc.
 * 
 * See the contributors.txt file in the distribution for a
 * full listing of individual contributors.
 * 
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 3.0 of
 * the License, or (at your option) any later version.
 * 
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * 
 * You should have received a copy of the GNU General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

#import "Label.h"


@implementation Label

@synthesize fontSize, color, text;

// This method is abstract method of indirectclass XMLEntity.
// So, this method must be overridden in subclass.
- (NSString *) elementName {
	return LABEL;
}


#pragma mark Delegate methods of NSXMLParser  

- (id)initWithXMLParser:(NSXMLParser *)parser elementName:(NSString *)elementName attributes:(NSDictionary *)attributeDict parentDelegate:(NSObject *)parent {
	if (self = [super init]) {
		componentId = [[attributeDict objectForKey:ID] intValue];
		fontSize = [[attributeDict objectForKey:FONT_SIZE] intValue];
		color = [[attributeDict objectForKey:COLOR] copy];
		text = [[attributeDict objectForKey:TEXT] copy];
		xmlParserParentDelegate = [parent retain];
		[parser setDelegate:self];
	}
	return self;
}

- (void)dealloc {
	[color release];
	[text release];
	[super dealloc];
}

@end
